### PR TITLE
Process bulk request refusals

### PR DESCRIPTION
**Pull request by AdamHawtin**
# Motivation and Context
We need a way of turning files of bulk refusals containing case ID's and refusal types into RM messages and feeding back any validation failures. The aim is also to make this a reusable 'bulk processing' framework so that ways of processing other file/message types are easy to add in the future.

# What has changed
* Add bulk processing module
* Add refusal processor and tests
* Close connections in the DB helper methods
* Update README

# How to test?
Build and push a tagged version of toolbox on this branch and terraform/apply using the branches attached to the card. Then run the bulk refusals processor and check it can successfully refuse any good rows and feeds back appropriate errors for any incorrect rows (bad refusal types, bad case ID's, bad file format etc.)

# Links
https://trello.com/c/I3pciSPB/1116-process-bulk-request-refusals-13

### CODE REVIEW

**Comment by ryangrundy7** [comment]
[Process Bulk request - Refusals (13)](https://trello.com/c/I3pciSPB/1116-process-bulk-request-refusals-13)

File: bulk_processing/bulk_processor.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    def process_rows(self, file_reader, success_file, error_file, error_detail_file):
+        error_count = 0
+        success_count = 0
+        for line_number, row in enumerate(file_reader, 2):
+            row_errors = self.find_row_validation_errors(line_number, row)
+            if row_errors:
+                error_count += len(row_errors)
+                self.write_row_errors_to_files(row, row_errors, error_file, error_detail_file)
+            else:
+                success_count += 1
```
Comment (NickGrantONS) [inline bulk_processing/bulk_processor.py:84]: why is the success count not the number of successful rows? Isn't this more like a success boolean than a count?

**Review by NickGrantX — APPROVED** [review]
Great work - looks fantastic

**Review by liamtoozer — APPROVED** [review]
Nice work! LGTM 👍

**Pull request merged**